src/Compilers/VisualBasic/Portable/Symbols/Source/SourceMemberMethodSymbol.vb

```diff
@@ -743,6 +743,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     diagBag)
             End If
 
+            Select Case ContainingType.TypeKind
+                Case TypeKind.Interface, TypeKind.Structure, TypeKind.Enum, TypeKind.Delegate
+                    ' Handles clause is invalid in this context. 
+                    Return Nothing
+
+                Case TypeKind.Class, TypeKind.Module
+                    ' Valid context
+
+                Case Else
+                    Debug.Assert(False)
```

Comment:
If you used the assert in ```ExceptionUtilities.UnexpectedValue``` it would include the type and value in the assert message.  Also, do you want to return ```Nothing```?